llvm/include/llvm/IR/IntrinsicsAMDGPU.td

```diff
@@ -12,11 +12,16 @@
 
 def global_ptr_ty : LLVMQualPointerType<1>;
 
+// The amdgpu-no-* attributes (ex amdgpu-no-workitem-id-z) typically inferred
+// by the backend cause whole-program undefined behavior when violated, such as
+// by causing all other preload register intrinsics to return arbitrarily incorrect
+// values. Outside of such IR-level UB, these preloaded registers are always set
```

Comment:
This logic only really applies in the entry functions

Reply:
True. That being said, if I want workitem ID Y and I'm being called from a amdgpu-no-workitem-id-y function, that function has to invoke undefined behavior in order to carry out the function call ... or the calling convention preserves the preloaded registers and those become incorrectly laid out like in the entry function.

I'll probably want a better way to word this